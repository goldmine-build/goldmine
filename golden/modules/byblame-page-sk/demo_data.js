export const fakeNow = Date.parse('2019-11-08T00:00:00Z');

export const canvaskit = {
  data: [],
};

export const gm = {
  data: [{
    groupID: '4edb719f1bc49bae585ff270df17f08039a96b6c:252cdb782418949651cc5eb7d467c57ddff3d1c7:a1050ed2b1120613d9ae9587e3c0f4116e17337f:3f7c865936cc808af26d88bc1f5740a29cfce200:05f6a01bf9fd25be9e5fff4af5505c3945058b1d',
    nDigests: 1,
    nTests: 1,
    affectedTests: [{
      test: 'complexclip_bw_layer',
      num: 1,
      sample_digest: '457875b610908dde8bfc5f45c907eb62',
    }],
    commits: [{
      commit_time: 1573171214,
      hash: '05f6a01bf9fd25be9e5fff4af5505c3945058b1d',
      author: 'Elisa (elisa@example.com)',
      message: 'Revert Revert Reland Update commits to have subject information',
    }, {
      commit_time: 1573171074,
      hash: '3f7c865936cc808af26d88bc1f5740a29cfce200',
      author: 'Iris (iris@example.com)',
      message: 'Revert Reland Update commits to have subject information',
    }, {
      commit_time: 1573170075,
      hash: 'a1050ed2b1120613d9ae9587e3c0f4116e17337f',
      author: 'Daniel (daniel@example.com)',
      message: 'Reland Update commits to have subject information',
    }, {
      commit_time: 1573179814,
      hash: '252cdb782418949651cc5eb7d467c57ddff3d1c7',
      author: 'Elisa (elisa@example.com)',
      message: 'Revert Update commits to have subject information',
    }, {
      commit_time: 1573179564,
      hash: '4edb719f1bc49bae585ff270df17f08039a96b6c',
      author: 'Joe (joe@example.com)',
      message: 'Update commits to have subject information',
    }],
  }, {
    groupID: '4edb719f1bc49bae585ff270df17f08039a96b6c:252cdb782418949651cc5eb7d467c57ddff3d1c7',
    nDigests: 7,
    nTests: 7,
    affectedTests: [{
      test: 'aarectmodes',
      num: 1,
      sample_digest: 'c6476baec94eb6a5071606575318e4df',
    }, {
      test: 'aaxfermodes',
      num: 1,
      sample_digest: '4acfd6b3a3943cc5d75cd22e966ae6f1',
    }, {
      test: 'hairmodes',
      num: 1,
      sample_digest: 'f9e20c63b5ce3b58d9b6a90fa3e7224c',
    }, {
      test: 'imagefilters_xfermodes',
      num: 1,
      sample_digest: '47644613317040264fea6fa815af32e8',
    }, {
      test: 'lattice2',
      num: 1,
      sample_digest: '16e41798ecd59b1523322a57b49cc17f',
    }, {
      test: 'xfermodes',
      num: 1,
      sample_digest: '8fbee03f794c455c4e4842ec2736b744',
    }, {
      test: 'xfermodes3',
      num: 1,
      sample_digest: 'fed2ff29abe371fc0ec1b2c65dfb3949',
    }],
    commits: [{
      commit_time: 1573149814,
      hash: '252cdb782418949651cc5eb7d467c57ddff3d1c7',
      author: 'Elisa (elisa@example.com)',
      message: 'Reland Update commits to have subject information',
    }, {
      commit_time: 1573149564,
      hash: '4edb719f1bc49bae585ff270df17f08039a96b6c',
      author: 'Joe (joe@example.com)',
      message: 'Revert Update commits to have subject information',
    }],
  }, {
    groupID: '73a722ce97ad935f936a4c7512b6724c41e0ce4e',
    nDigests: 41,
    nTests: 1,
    affectedTests: [{
      test: 'skottie_colorize',
      num: 41,
      sample_digest: '024ce342b014c6fdb000f7c18d6d0775',
    }],
    commits: [{
      commit_time: 1572978684,
      hash: '73a722ce97ad935f936a4c7512b6724c41e0ce4e',
      author: 'Bob (bob@example.com)',
      message: 'Update commits to have subject information 4',
    }],
  }, {
    groupID: '85c3d68f2539ed7a1e71f6c9d12baaf9e6be59d8',
    nDigests: 51,
    nTests: 31,
    affectedTests: null,
    commits: [{
      commit_time: 1572899861,
      hash: '85c3d68f2539ed7a1e71f6c9d12baaf9e6be59d8',
      author: 'Alice (alice@example.com)',
      message: 'Update commits to have subject information 3',
    }],
  }, {
    groupID: '7da048b5e8f17374bcd5baf48539eaa7ebe40e5c',
    nDigests: 13,
    nTests: 3,
    affectedTests: [{
      test: 'shadow_utils',
      num: 5,
      sample_digest: '134d3c4fd609cd7f2e9cca43d78aa5d3',
    }, {
      test: 'shadow_utils_gray',
      num: 4,
      sample_digest: '292eb1e5b5860ba278ffa73efc9dd7c1',
    }, {
      test: 'shadow_utils_occl',
      num: 4,
      sample_digest: '918df0cc65d1b3ae7b8d8041afa40635',
    }],
    commits: [{
      commit_time: 1572442816,
      hash: '7da048b5e8f17374bcd5baf48539eaa7ebe40e5c',
      author: 'Frank (frank@example.com)',
      message: 'Update commits to have subject information 2',
    }],
  }, {
    groupID: '342fbc54844d0d3fc9d20e20b45115db1e33395b',
    nDigests: 1,
    nTests: 1,
    affectedTests: [{
      test: 'dftext_blob_persp',
      num: 1,
      sample_digest: '42a7bfc2d825412dfb82f6e3a12106c2',
    }],
    commits: [{
      commit_time: 1571959019,
      hash: '342fbc54844d0d3fc9d20e20b45115db1e33395b',
      author: 'Liam (liam@example.com)',
      message: 'Oh my goodness, this is an incredibly long subject line. '
        + 'How will it every fit on the page? I hope this draws ok. Some webpages '
        + 'do not work when the content is way bigger than anticipated. I hope this is '
        + 'not one of those websites',
    }],
  }],
};

export const svg = {
  data: [{
    groupID: 'd2c67f44f8c2351e60e6ee224a060e916cd44f34',
    nDigests: 12,
    nTests: 12,
    affectedTests: null,
    commits: [{
      commit_time: 1571948193,
      hash: 'd2c67f44f8c2351e60e6ee224a060e916cd44f34',
      author: 'Alice (alice@example.com)',
      message: 'Update commits to have subject information',
    }],
  }, {
    groupID: '05f6a01bf9fd25be9e5fff4af5505c3945058b1d',
    nDigests: 2,
    nTests: 1,
    affectedTests: [{
      test: 'A_large_blank_world_map_with_oceans_marked_in_blue.svg',
      num: 2,
      sample_digest: '3c62b1dd009bb18bbb84d862cbc3652c',
    }],
    commits: [{
      commit_time: 1573151114,
      hash: '05f6a01bf9fd25be9e5fff4af5505c3945058b1d',
      author: 'Elisa (elisa@example.com)',
      message: 'Update commits to have subject information',
    }],
  }, {
    groupID: 'd84dd4babb71796ee194fa1913150d86d6aa643b',
    nDigests: 2,
    nTests: 1,
    affectedTests: [{
      test: 'ynev.svg',
      num: 2,
      sample_digest: '80a7bcb4b51ad1aa876aa2144d0648bb',
    }],
    commits: [{
      commit_time: 1573048663,
      hash: 'd84dd4babb71796ee194fa1913150d86d6aa643b',
      author: 'Henry (henry@example.com)',
      message: 'Update commits to have subject information',
    }],
  }, {
    groupID: '3f7c865936cc808af26d88bc1f5740a29cfce200:05f6a01bf9fd25be9e5fff4af5505c3945058b1d',
    nDigests: 1,
    nTests: 1,
    affectedTests: [{
      test: 'A_large_blank_world_map_with_oceans_marked_in_blue.svg',
      num: 1,
      sample_digest: '02fc74ad862a6041e9ea35769a67af20',
    }],
    commits: [{
      commit_time: 1573151114,
      hash: '05f6a01bf9fd25be9e5fff4af5505c3945058b1d',
      author: 'Elisa (elisa@example.com)',
      message: 'Revert: Update commits to have subject information',
    }, {
      commit_time: 1573151074,
      hash: '3f7c865936cc808af26d88bc1f5740a29cfce200',
      author: 'Iris (iris@example.com)',
      message: 'Update commits to have subject information',
    }],
  }, {
    groupID: 'e1e197186238d8d304a39db9f94258d9584a8973',
    nDigests: 1,
    nTests: 1,
    affectedTests: [{
      test: 'gallardo.svg',
      num: 1,
      sample_digest: 'ed591512ff68fa088ec9aac3bb5d760d',
    }],
    commits: [{
      commit_time: 1572460288,
      hash: 'e1e197186238d8d304a39db9f94258d9584a8973',
      author: 'Iris (iris@example.com)',
      message: 'Update commits to have subject information',
    }],
  }],
};


export const trstatus = {
  ok: false,
  firstCommit: {
    commit_time: 1572357082,
    hash: 'ee08d523f60a04499c9023a349ef8174ab301f8f',
    author: 'Alice (alice@example.com)',
    message: 'Update commits to have subject information',
  },
  lastCommit: {
    commit_time: 1573598625,
    hash: '9501212cd0580acfed85a90c3a16b81847fde482',
    author: 'Bob (bob@example.com)',
    message: 'Revert Update commits to have subject information',
  },
  totalCommits: 256,
  filledCommits: 256,
  corpStatus: [{
    name: 'canvaskit',
    ok: false,
    minCommitHash: '',
    untriagedCount: 0,
    negativeCount: 0,
  }, {
    name: 'gm',
    ok: false,
    minCommitHash: '',
    untriagedCount: 114,
    negativeCount: 1494,
  }, {
    name: 'svg',
    ok: false,
    minCommitHash: '',
    untriagedCount: 18,
    negativeCount: 21,
  }],
};
