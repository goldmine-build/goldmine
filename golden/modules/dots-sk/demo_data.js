// The trace below is based on a subset of a real trace found in Skia Gold.
// This is what the dots diagram should look like (trace length = 20):
//
//   +--------------------+
//   |**765334433332211100|
//   |   11-1-1100--0000  |
//   |      22221111110000|
//   +--------------------+
//
// Where the numbers represent different colors, and a star (*) represents the
// special color used for unique digests in excess of MAX_UNIQUE_DIGESTS.
//
// Additionally, The numbers above correspond to traces.traces[i].data[j].s.

export const traces = {
  "tileSize": 20,
  "traces": [{
    "data": [{
      "x": 0,
      "y": 0,
      // Note: the backend tops out at s=8, which equals MAX_UNIQUE_DIGESTS.
      "s": 9
    }, {
      "x": 1,
      "y": 0,
      "s": 8
    }, {
      "x": 2,
      "y": 0,
      "s": 7
    }, {
      "x": 3,
      "y": 0,
      "s": 6
    }, {
      "x": 4,
      "y": 0,
      "s": 5
    }, {
      "x": 5,
      "y": 0,
      "s": 3
    }, {
      "x": 6,
      "y": 0,
      "s": 3
    }, {
      "x": 7,
      "y": 0,
      "s": 4
    }, {
      "x": 8,
      "y": 0,
      "s": 4
    }, {
      "x": 9,
      "y": 0,
      "s": 3
    }, {
      "x": 10,
      "y": 0,
      "s": 3
    }, {
      "x": 11,
      "y": 0,
      "s": 3
    }, {
      "x": 12,
      "y": 0,
      "s": 3
    }, {
      "x": 13,
      "y": 0,
      "s": 2
    }, {
      "x": 14,
      "y": 0,
      "s": 2
    }, {
      "x": 15,
      "y": 0,
      "s": 1
    }, {
      "x": 16,
      "y": 0,
      "s": 1
    }, {
      "x": 17,
      "y": 0,
      "s": 1
    }, {
      "x": 18,
      "y": 0,
      "s": 0
    }, {
      "x": 19,
      "y": 0,
      "s": 0
    }],
    "label": ",alpha=first-trace,beta=hello,gamma=world,",
    "params": {
      "alpha": "first-trace",
      "beta": "hello",
      "gamma": "world",
    }
  }, {
    "data": [{
      "x": 3,
      "y": 1,
      "s": 1
    }, {
      "x": 4,
      "y": 1,
      "s": 1
    }, {
      "x": 6,
      "y": 1,
      "s": 1
    }, {
      "x": 8,
      "y": 1,
      "s": 1
    }, {
      "x": 9,
      "y": 1,
      "s": 1
    }, {
      "x": 10,
      "y": 1,
      "s": 0
    }, {
      "x": 11,
      "y": 1,
      "s": 0
    }, {
      "x": 14,
      "y": 1,
      "s": 0
    }, {
      "x": 15,
      "y": 1,
      "s": 0
    }, {
      "x": 16,
      "y": 1,
      "s": 0
    }, {
      "x": 17,
      "y": 1,
      "s": 0
    }],
    "label": ",alpha=second-trace,beta=foo,gamma=bar,",
    "params": {
      "alpha": "second-trace",
      "beta": "foo",
      "gamma": "bar",
    }
  }, {
    "data": [{
      "x": 6,
      "y": 2,
      "s": 2
    }, {
      "x": 7,
      "y": 2,
      "s": 2
    }, {
      "x": 8,
      "y": 2,
      "s": 2
    }, {
      "x": 9,
      "y": 2,
      "s": 2
    }, {
      "x": 10,
      "y": 2,
      "s": 1
    }, {
      "x": 11,
      "y": 2,
      "s": 1
    }, {
      "x": 12,
      "y": 2,
      "s": 1
    }, {
      "x": 13,
      "y": 2,
      "s": 1
    }, {
      "x": 14,
      "y": 2,
      "s": 1
    }, {
      "x": 15,
      "y": 2,
      "s": 1
    }, {
      "x": 16,
      "y": 2,
      "s": 0
    }, {
      "x": 17,
      "y": 2,
      "s": 0
    }, {
      "x": 18,
      "y": 2,
      "s": 0
    }, {
      "x": 19,
      "y": 2,
      "s": 0
    }],
    "label": ",alpha=third-trace,beta=baz,gamma=qux,",
    "params": {
      "alpha": "second-trace",
      "beta": "baz",
      "gamma": "qux",
    }
  }],
  "digests": [{
    "digest": "ce0a9d2b546b25e00e39a33860cb72b6",
    "status": "untriaged"
  }, {
    "digest": "34e87ca0f753cf4c884fa01af6c08be9",
    "status": "positive"
  }, {
    "digest": "8ee9a2c61e9f12e6243f07423302f26a",
    "status": "untriaged"
  }, {
    "digest": "6174fc17b06e6ff9e383da3f6952ad68",
    "status": "positive"
  }, {
    "digest": "dcccd6998b47f60ab28dcff17ae57ed2",
    "status": "untriaged"
  }, {
    "digest": "92d9faf80a25750629118018716387df",
    "status": "untriaged"
  }, {
    "digest": "1bc4771dcee95d97b2758a1e1945cc40",
    "status": "untriaged"
  }, {
    "digest": "a9f4c341392618fad087060a0e69f170",
    "status": "untriaged"
  }, {
    "digest": "9522095d651fd5e6572904a1c13fb91c",
    "status": "untriaged"
  }, {
    "digest": "8ad66f50b755d82cd1c08b22e984bbef",
    "status": "untriaged"
  }]
};

export const commits = [{
  "commit_time": 1576186931,
  "hash": "46a331b93f54d8b3bce88792dd8679beef11a751",
  "author": "Alpha (alpha@example.com)"
}, {
  "commit_time": 1576186932,
  "hash": "1521e6b24c19f30eda383bb00b26862894ae9182",
  "author": "Beta (beta@example.com)"
}, {
  "commit_time": 1576189965,
  "hash": "f46d5ca49221113497d41e8f2a3c0c59151f4010",
  "author": "Gamma (gamma@example.com)"
}, {
  "commit_time": 1576190315,
  "hash": "dcd8e9389d8aa79a389aebad570d340ec012f367",
  "author": "Beta (beta@example.com)"
}, {
  "commit_time": 1576191335,
  "hash": "2fc9fa6d08df3b12c764d88f4458d28d4352de9b",
  "author": "Epsilon (epsilon@example.com)"
}, {
  "commit_time": 1576192005,
  "hash": "4d3b4a1bf31afb9d50ac84221a5852fea29a30df",
  "author": "Beta (beta@example.com)"
}, {
  "commit_time": 1576197935,
  "hash": "0678df30b5a56375ff6a4c21e6f0ecadb3493b7c",
  "author": "Delta (delta@example.com)"
}, {
  "commit_time": 1576200535,
  "hash": "39cdc37bdd0fd63556357a86b43f83fa4211ce0f",
  "author": "Beta (beta@example.com)"
}, {
  "commit_time": 1576211413,
  "hash": "252a03454d382b387c8f42aa75cfd63756816713",
  "author": "Beta (beta@example.com)"
}, {
  "commit_time": 1576213625,
  "hash": "415bce89a49abb6f53b2d3634159f6d304c8c8b5",
  "author": "Beta (beta@example.com)"
}, {
  "commit_time": 1576213773,
  "hash": "7fb7134e7d946d80741f779cfdb10cfd40a1f7a3",
  "author": "Beta (beta@example.com)"
}, {
  "commit_time": 1576246423,
  "hash": "d0840ecf583171e55025d2808dba017910b7a54f",
  "author": "Zeta (zeta@example.com)"
}, {
  "commit_time": 1576255153,
  "hash": "81b98978bced13406df91c2f5917cc2b82772f1e",
  "author": "Eta (Eta@example.com)"
}, {
  "commit_time": 1576258473,
  "hash": "a6069a154d66b2620bea1907b0eebf5d1afd02e7",
  "author": "Theta (Theta@example.com)"
}, {
  "commit_time": 1576260733,
  "hash": "1c5be7b19707c54ff859aa9f834a92e14d6ab5b9",
  "author": "Epsilon (epsilon@example.com)"
}, {
  "commit_time": 1576260973,
  "hash": "ab51c2ce0884a2bb1693d0f15d9eb674800e18ba",
  "author": "Iota (iota@example.com)"
}, {
  "commit_time": 1576262533,
  "hash": "c9b4d279d235c8db48875f0d0854bfe25c631ff6",
  "author": "Kappa (kappa@example.com)"
}, {
  "commit_time": 1576264923,
  "hash": "1cc767bd0d915bd0f3f5b40dcb282367c9fd9271",
  "author": "Gamma (gamma@example.com)"
}, {
  "commit_time": 1576265403,
  "hash": "a072b7b2758d644fbd5483a9716f581d270c7560",
  "author": "Lambda (lambda@example.com)"
}, {
  "commit_time": 1576265853,
  "hash": "17e7dfa37734347215f0b6bacb72c06ec85dbfdc",
  "author": "Epsilon (epsilon@example.com)"
}];
