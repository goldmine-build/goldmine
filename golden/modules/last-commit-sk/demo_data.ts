import {StatusResponse} from '../rpc_types';

export const exampleStatusData: StatusResponse = {
  ok: true,
  firstCommit: {
    id: 'alphace63',
    commit_time: 1597819864,
    hash: 'ce63f507336fa86278204dfac2b9a9546f81ab96',
    author: 'Alpha Beta (alphabeta@example.com)',
    message: "Document how to size IV's child correctly, after seeing confusion in Github issues (#64100)",
    cl_url: '',
  },
  lastCommit: {
    id: 'foxtrota828',
    commit_time: 1598983079,
    hash: 'a8281e31afa9dddfa0764f59128c3a2360c48f49',
    author: 'Foxtrot Delta (foxtrot.delta@example.com)',
    message: 'Mark large_image_changer tests as not flaky (#65033)',
    cl_url: '',
  },
  totalCommits: 200,
  filledCommits: 200,
  corpStatus: [{
    name: 'flutter', ok: true, minCommitHash: '', untriagedCount: 0, negativeCount: 0,
  }],
};
