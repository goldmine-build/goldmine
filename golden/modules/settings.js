/**
 * This file lists helpers for returning global constants in Gold. These should not change once
 * an instance has booted up, so they need not be returned via JSON requests.
 *
 * Settings are expected to be in window.GoldSettings and the functions here are nice helpers
 * for that, so as to demystify "where do these values come from?"
 */

/**
 * @return {string}
 */
export function title() {
  return window.GoldSettings && window.GoldSettings.title;
}

/**
 * @return {string}
 */
export function defaultCorpus() {
  return window.GoldSettings && window.GoldSettings.defaultCorpus;
}

/**
 * @return {string}
 */
export function baseRepoURL() {
  return window.GoldSettings && window.GoldSettings.baseRepoURL;
}

/**
 * @return {string}
 */
export function codeReviewURLTemplate() {
  return window.GoldSettings && window.GoldSettings.crsTemplate;
}

/**
 * @param newSettings {Object}
 */
export function testOnlySetSettings(newSettings) {
  window.GoldSettings = newSettings;
}
