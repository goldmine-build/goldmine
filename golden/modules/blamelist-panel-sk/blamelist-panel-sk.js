/**
 * @module module/blamelist-panel-sk
 * @description <h2><code>blamelist-panel-sk</code></h2>
 *
 * A list of commits and authors. If the list is too long, the first several will be shown.
 *
 * This should typically go into some sort of dialog to show the user.
 */
import { define } from 'elements-sk/define';
import { html } from 'lit-html';
import { diffDate } from 'common-sk/modules/human';
import { ElementSk } from '../../../infra-sk/modules/ElementSk';
import { truncateWithEllipses } from '../common';
import { baseRepoURL } from '../settings';

const maxCommitsToDisplay = 15;

const template = (ele) => html`
<h2>Commits:</h2>
<table>
  ${ele._commits.slice(0, maxCommitsToDisplay).map(commitRow)}
</table>
<div>
  ${ele._commits.length > maxCommitsToDisplay ? '...and other commits.' : ''}
</div>`;

const commitRow = (c) => html`
<tr>
  <td title=${c.author}>${truncateWithEllipses(c.author, 20)}</td>
  <td title=${new Date(c.commit_time * 1000)}>
   ${diffDate(c.commit_time * 1000)}
  </td>
  <td><a href=${commitHref(c)}>${c.hash && c.hash.substring(0, 8)}</a></td>
  <td title=${c.message}>${truncateWithEllipses(c.message || '', 80)}</td>
</tr>
`;

const commitHref = (commit) => {
  // TODO(kjlubick) deduplicate with by-blame-sk
  const repo = baseRepoURL();
  if (!repo) {
    throw new DOMException('repo not set in settings');
  }
  if (repo.indexOf('github.com') !== -1) {
    return `${repo}/commit/${commit.hash}`;
  }
  return `${repo}/+/${commit.hash}`;
};

define('blamelist-panel-sk', class extends ElementSk {
  constructor() {
    super(template);

    this._commits = [];
  }

  connectedCallback() {
    super.connectedCallback();
    this._render();
  }

  /**
   * @prop commits {Array<Object>} the commits to show. The objects should have string fields:
   *   author, message, hash and a field commit_time that is the number of seconds since the epoch.
   *   See frontend.Commit on the server side for more.
   */
  get commits() { return this._commits; }

  set commits(arr) {
    this._commits = arr;
    this._render();
  }
});
