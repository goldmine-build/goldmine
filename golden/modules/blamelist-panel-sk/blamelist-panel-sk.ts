/**
 * @module module/blamelist-panel-sk
 * @description <h2><code>blamelist-panel-sk</code></h2>
 *
 * A list of commits and authors. If the list is too long, the first several will be shown.
 *
 * This should typically go into some sort of dialog to show the user.
 */
import { define } from 'elements-sk/define';
import { html } from 'lit-html';
import { diffDate } from 'common-sk/modules/human';
import { ElementSk } from '../../../infra-sk/modules/ElementSk';
import { truncateWithEllipses } from '../common';
import { baseRepoURL } from '../settings';

const maxCommitsToDisplay = 15;

const template = (ele: BlamelistPanelSk) => html`
<h2>Commits:</h2>
<table>
  ${ele.commits.slice(0, maxCommitsToDisplay).map(commitRow)}
</table>
<div>
  ${ele.commits.length > maxCommitsToDisplay ? '...and other commits.' : ''}
</div>`;

const commitRow = (c: Commit) => html`
<tr>
  <td title=${c.author}>${truncateWithEllipses(c.author, 20)}</td>
  <td title=${new Date(c.commit_time * 1000)}>
   ${diffDate(c.commit_time * 1000)}
  </td>
  <td><a href=${commitHref(c)}>${c.hash && c.hash.substring(0, 8)}</a></td>
  <td title=${c.message}>${truncateWithEllipses(c.message || '', 80)}</td>
</tr>
`;

const commitHref = (commit: Commit) => {
  // TODO(kjlubick): Deduplicate with by-blame-sk.
  const repo = baseRepoURL();
  if (!repo) {
    throw new DOMException('repo not set in settings');
  }
  if (repo.indexOf('github.com') !== -1) {
    return `${repo}/commit/${commit.hash}`;
  }
  return `${repo}/+show/${commit.hash}`;
};

/**
 * Represents a Git commit.
 * 
 * Client-side equivalent of frontend.Commit Go type.
 */
export interface Commit {
  readonly hash: string;
  readonly author: string;
  readonly message: string;
  readonly commit_time: number;
};

export class BlamelistPanelSk extends ElementSk {
  private _commits: Commit[] = [];

  constructor() {
    super(template);
  }

  connectedCallback() {
    super.connectedCallback();
    this._render();
  }

  get commits(): Commit[] { return this._commits; }

  set commits(commits: Commit[]) {
    this._commits = commits;
    this._render();
  }
};

define('blamelist-panel-sk', BlamelistPanelSk);
