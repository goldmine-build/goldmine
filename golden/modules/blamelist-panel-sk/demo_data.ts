import { Commit } from './blamelist-panel-sk';

export const fakeNow = Date.parse('2020-03-22T00:00:00.000Z');

export const blamelist19: Commit[] = [
  {
    commit_time: 1584835000,
    hash: 'dded3c7506efc5635e60ffb7a908cbe8f1f028f1',
    author: 'Alfa (alfa@example.com)',
    message: 'Update provisioning_profile to unbreak iOS since cert refresh',
    is_cl: false,
  },
  {
    commit_time: 1584834200,
    hash: '44fc53b7f5d8390d7ae228d5fa8bac0b225ed2b9',
    author: 'Bravo (bravo@example.com)',
    message: 'Add BGR_10A2 support to Ganesh',
    is_cl: false,
  },
  {
    commit_time: 1584834100,
    hash: 'c50f4e9399cc7ae72b05eb641b50a7cd06d5af82',
    author: 'Charlie (charlie@example.com)',
    message: 'Simplify GrTextBlob::flush',
    is_cl: false,
  },
  {
    commit_time: 1584834000,
    hash: '9145f784f3261f227846e5b08dc2691a888b113c',
    author: 'Delta (deltadelta@example.com)',
    message: 'Implement D3D copySurface.',
    is_cl: false,
  },
  {
    commit_time: 1584824000,
    hash: '02b91385f58b84dc14cb6f81478b36d7fcb0d24b',
    author: 'Echo Foxtrot (ef@example.com)',
    message: 'Rename flush -> addOp',
    is_cl: false,
  },
  {
    commit_time: 1584814000,
    hash: 'a1e90aba441b0e68a07e3ab6660cd673335a76f5',
    author: 'Golf Golf Golf Golf Golf Golf Golf(golf@example.com)',
    message: 'Turn on Vulkan bots for Galaxy S20.',
    is_cl: false,
  },
  {
    commit_time: 1584804000,
    hash: 'b2b9fbc5675c46cb3af30ccd9b6e20bc57ec04c8',
    author: 'Hotel India (hotel.ind@example.com)',
    message: 'Compare all fields in SkSL::Layout::operator==',
    is_cl: false,
  },
  {
    commit_time: 1584800000,
    hash: 'f1792cde0b2b7b612cac6aa5b9e7a8ce16b9b2cb',
    author: 'Hotel India (hotel.ind@example.com)',
    message: 'Use constant swizzle syntax in GrDrawVerticesOp',
    is_cl: false,
  },
  {
    commit_time: 1584780000,
    hash: 'fc75d5c52da617b06c7566c9e389587edde3f284',
    author: 'Juliett Kilo (julia.kilo@example.com)',
    message: '[skottie] Brightness and Contrast effect',
    is_cl: false,
  },
  {
    commit_time: 1584760000,
    hash: 'e15088801e9722d9a320463554a3a21839213b48',
    author: 'Mike Lima (lima@example.com)',
    message: 'detect failed matrix update in SkDraw::drawAtlas()',
    is_cl: false,
  },
  {
    commit_time: 1584760000,
    hash: '548de7451e752ce0d6fd842c1bb4f04af4c0afdc',
    author: 'Hotel India (hotel.ind@example.com)',
    message: 'Change Marker IDs to be strings',
    is_cl: false,
  },
  {
    commit_time: 1584740000,
    hash: '7676f4ecf3ec2f544701465faca1d33af1489822',
    author: 'Zulu (zulu@example.com)',
    message: 'Remove SkFrontBufferedStream',
    is_cl: false,
  },
  {
    commit_time: 1584720000,
    hash: '4baa7326ccfbfaf66e25d91628378536d4999999',
    author: 'Papa November (pnov@example.com)',
    message: '[infra] Add POC task driver',
    is_cl: false,
  },
  {
    commit_time: 1584700000,
    hash: 'f5132a05c893a86f8bf26bcf3253985d9973fea2',
    author: 'Quebec (quebec@example.com)',
    message: "Reland \"Optimize GrTessellatePathOp's code to emit inner tri",
    is_cl: false,
  },
  {
    commit_time: 1584600005,
    hash: '3d599d16ecd8009dfb185522e87c5c1cf8a47057',
    author: 'skia-recreate-skps (skia-recreate-skps@example.com)',
    message: 'Update Go Deps',
    is_cl: false,
  },
  {
    commit_time: 1584600005,
    hash: '7a0517f8bdc0d168565a2944f0c14db06084384d',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll third_party/externals/angle2 3cb9c4bee9b3..4395170e6091',
    is_cl: false,
  },
  {
    commit_time: 1584600003,
    hash: 'd18100afb62401b49e0b3a5c87ed8fa006e8d3a6',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll ../src 9781ff27c9e9..78824aa9d99f (468 commits)',
    is_cl: false,
  },
  {
    commit_time: 1584600002,
    hash: '2ac6fa0ae796b7f050bd8e8ad60890427a4ec15e',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll third_party/externals/swiftshader 60aa34a990fa..2717702',
    is_cl: false,
  },
  {
    commit_time: 1584600001,
    hash: '667edf14ad72966ec36aa6cd705b98cb7d7eee28',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll third_party/externals/dawn 00b90ea83262..88f2ec853f80 (',
    is_cl: false,
  },
];

export const clBlamelist: Commit[] = [
  {
    commit_time: 1584600001,
    hash: '12345',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll third_party/externals/dawn 00b90ea83262..88f2ec853f80 (',
    is_cl: true,
  },
  {
    commit_time: 1584835000,
    hash: 'dded3c7506efc5635e60ffb7a908cbe8f1f028f1',
    author: 'Alfa (alfa@example.com)',
    message: 'Update provisioning_profile to unbreak iOS since cert refresh',
    is_cl: false,
  },
];
