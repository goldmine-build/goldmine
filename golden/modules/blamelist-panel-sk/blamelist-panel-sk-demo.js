import './index';
import { $$ } from 'common-sk/modules/dom';
import { blamelist19, fakeNow } from './demo_data';
import { testOnlySetSettings } from '../settings';

Date.now = () => fakeNow;

testOnlySetSettings({
  baseRepoURL: 'https://github.com/example/example',
});

let ele = document.createElement('blamelist-panel-sk');
ele.commits = blamelist19.slice(0, 1);
$$('#single_commit').appendChild(ele);

ele = document.createElement('blamelist-panel-sk');
ele.commits = blamelist19.slice(0, 3);
$$('#some_commits').appendChild(ele);

ele = document.createElement('blamelist-panel-sk');
ele.commits = blamelist19;
$$('#many_commits').appendChild(ele);
