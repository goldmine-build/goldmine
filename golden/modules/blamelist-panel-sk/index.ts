import './blamelist-panel-sk';
import './blamelist-panel-sk.scss';
