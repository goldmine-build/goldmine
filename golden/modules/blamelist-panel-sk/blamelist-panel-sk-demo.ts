import './index';
import { $$ } from 'common-sk/modules/dom';
import { BlamelistPanelSk } from './blamelist-panel-sk';
import { blamelist19, clBlamelist, fakeNow } from './demo_data';
import { testOnlySetSettings } from '../settings';

Date.now = () => fakeNow;

testOnlySetSettings({
  baseRepoURL: 'https://github.com/example/example',
  crsTemplate: 'https://skia-review.googlesource.com/%s',
});

let ele = new BlamelistPanelSk();
ele.commits = blamelist19.slice(0, 2);
$$('#single_commit')!.appendChild(ele);

ele = new BlamelistPanelSk();
ele.commits = clBlamelist;
$$('#single_cl_commit')!.appendChild(ele);

ele = new BlamelistPanelSk();
ele.commits = blamelist19.slice(0, 4);
$$('#some_commits')!.appendChild(ele);

ele = new BlamelistPanelSk();
ele.commits = blamelist19;
$$('#many_commits')!.appendChild(ele);
