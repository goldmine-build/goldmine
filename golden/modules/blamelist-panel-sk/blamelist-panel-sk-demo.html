<!DOCTYPE html>
<html>
<head>
  <title>blamelist-panel-sk</title>
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body>
  <h1>blamelist-panel-sk with a single commit</h1>
  <div id="single_commit"></div>

  <h1>blamelist-panel-sk with some commits</h1>
  <div id="some_commits"></div>

  <h1>blamelist-panel-sk with many commits (truncated)</h1>
  <div id="many_commits"></div>
</body>
</html>
